src/framework/preferences/js/AuxBuilder.js

```diff
@@ -349,10 +341,17 @@ var fluid_2_0 = fluid_2_0 || {};
         return auxSchema;
     };
 
+    // Go over the aux schema to replace terms with their given values.
     fluid.prefs.expandSchema = function (schemaToExpand, indexes, topCommonOptions, elementCommonOptions, mappedDefaults) {
         var auxSchema = fluid.prefs.expandSchemaImpl(schemaToExpand);
         auxSchema.namespace = auxSchema.namespace || "fluid.prefs.created_" + fluid.allocateGuid();
 
+        var terms = fluid.get(auxSchema, "terms");
+        if (terms) {
+            delete auxSchema.terms;
+            fluid.set(auxSchema, ["terms", "terms"], terms);
```

Comment:
Thanks cindyli - perhaps we can add a comment in expandSchema explaining to readers what the strategy is and where the site is where this output structure is consumed. Also, this pull request requires a corresponding pull to update https://github.com/fluid-project/infusion-docs/blob/master/src/documents/AuxiliarySchemaForPreferencesFramework.md in infusion-docs